src/Model.php

```diff
@@ -776,33 +776,64 @@ public function setMulti(array $fields)
      *
      * @return mixed
      */
-    public function get(string $field = null)
+    public function get(string $fieldName = null)
     {
-        if ($field === null) {
+        if ($fieldName === null) {
             // Collect list of eligible fields
             $data = [];
             if ($this->only_fields) {
                 // collect data for actual fields
-                foreach ($this->only_fields as $field) {
-                    $data[$field] = $this->get($field);
+                foreach ($this->only_fields as $fieldName) {
+                    $data[$fieldName] = $this->get($fieldName);
                 }
             } else {
                 // get all fields
-                foreach ($this->getFields() as $field => $f) {
-                    $data[$field] = $this->get($field);
+                foreach ($this->getFields() as $fieldName => $field) {
+                    $data[$fieldName] = $this->get($fieldName);
                 }
             }
 
             return $data;
         }
 
-        $this->checkOnlyFieldsField($field);
+        // if field is not existing maybe it is a traversing link or aggregate function
+        if (!$this->hasField($fieldName)) {
+            // alias for count aggregate, consistent with Model scopes
+            if ($fieldName === '#') {
+                $fieldName = 'count(*)';
+            }
+
+            // traverses relationships and returns value from a referenced model, e.g contact/company/name
+            if (str_contains($fieldName, '/')) {
+                $refs = explode('/', $fieldName);
+
+                $fieldName = array_pop($refs);
+
+                return $this->ref(implode('/', $refs))->get($fieldName);
+            }
+
+            // aggregate function
+            if (str_contains($fieldName, '(')) {
+                $args = [];
+                if (preg_match('/^([\w ]+)\(([\w \*]*)/', $fieldName, $args)) {
```

Comment:
why parent if then?

Reply:
Looks cleaner

Reply:
no, do not test is possibly matching and then a better match :)

also, use regex more like `~^(\w+)\(([\w*]*)\)$~` (or `~^(\w+)\((.*)\)$~`, I do not know what should be allowed)